src/proto/streams/streams.rs

```diff
@@ -19,9 +19,9 @@ where
 }
 
 /// Reference to the stream state
-#[derive(Debug)]
 pub(crate) struct StreamRef<B, P>
 where
+    B: Buf,
```

Comment:
Just curious, why is `B: Buf` needed?

Reply:
@seanmonstar the `Drop` impl requires `B: Buf` because `Send<B, P>::send_reset` is bounded on `B: Buf`. 

```
error[E0599]: no method named `send_reset` found for type `proto::streams::send::Send<B, P>` in the current scope
   --> src/proto/streams/streams.rs:725:30
    |
725 |                 actions.send.send_reset(
    |                              ^^^^^^^^^^
    |
    = note: the method `send_reset` exists but the following trait bounds were not satisfied:
            `B : bytes::Buf`
```
if the `Drop` impl is bounded on `B: Buf`, then the struct definition must also be bounded.

Reply:
Aww, shucks. Very well.

Reply:
I think, for extra credit, someone could go in and make the impl on `Send` only need `B: Buf` for `send_data`, and have all the other methods be in an impl without that bounds, but not a big deal.

Reply:
@seanmonstar started looking into that but it looks like there are a few other things bounded on `Buf` so it may not be quite that simple - at the very least, probably another PR.

Reply:
@seanmonstar WDYT of 5079b35fa9634bd879cf2875a8475656b41a08b3?

Reply:
Neat!